### PR TITLE
Load static assets uncached and purge automation

### PR DESCRIPTION
**Pull request by rachelwhitton**
## Summary

**[Caching: Advanced Topics](https://pantheon.io/docs/caching-advanced-topics#manually-expiring-cache-for-static-assets)**  Address a common pain point that frequently adds tension and confusion in early stages of the customer journey 

* provide example custom script for loading uncached static assets 
* provide example quicksilver script to auto purge after a given workflow such as `deploy`
* [ ] Todo: consider inclusion of additional QS example script that uses pantheon_clear_edge_all(); combined with passthru('drush cc all'); to achieve purging all caches automatically after deploying to live env (vs purging 1 file alone via pantheon_clear_edge_paths)

Attribution for the example scripts should go to @kyletaylored (thank you so much Kyle!)

cc @joebortelpantheon 

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | [Live](https://pantheon.io/docs) | [pr-6892](https://pr-6892--pantheon-docs.my.pantheonfrontend.website/docs) | Difference |
| --- | --- | --- | --- |
| Performance Score | `56.99999999999999` | `62` | `5` |
| Best Practice Score | `79` | `79` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-6892` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/31751/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/31751/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)


[Preview Environment](https://pr-6892--pantheon-docs.my.pantheonfrontend.website/docs)

File: source/content/caching-advanced-topics.md [hunk 1]
```diff
@@ -52,6 +52,21 @@ For CSS or JavaScript changes, Drupal and WordPress each offer methods to ensure
 
 Dev and Multidev environments do not cache static assets.
 
+Another solution to consider when debugging issues related to stale static assets is to load the file via a custom PHP script that sets a `no-cache` header value, for example: 
+
+```php
+<?php
+
+// Punch through the Pantheon edge cache.
```
Comment (ari-gold) [inline source/content/caching-advanced-topics.md:60]: Suggest changing to 
// Set custom cache-control header

File: source/content/caching-advanced-topics.md [hunk 2]
```diff
@@ -52,6 +52,21 @@ For CSS or JavaScript changes, Drupal and WordPress each offer methods to ensure
...
+header("Cache-Control: no-cache, must-revalidate");
```
Comment (ari-gold) [inline source/content/caching-advanced-topics.md:61]: Might want to have the sample be to cache for one day, or maybe two different lines (one with no-cache, one with a custom TTL) and they can uncomment the one they want

**Comment by rachelwhitton** [comment]
todo: consider inclusion of additional QS example script that uses `pantheon_clear_edge_all();` combined with `passthru('drush cc all');` to achieve purging **all** caches automatically after deploying to live env (vs purging 1 file alone via `pantheon_clear_edge_paths`)

File: source/content/caching-advanced-topics.md [hunk 3]
```diff
@@ -52,6 +52,47 @@ For CSS or JavaScript changes, Drupal and WordPress each offer methods to ensure
...
+
+// Set JSON header
+header('Content-Type: application/json');
+
+// Echo file content
+echo file_get_contents("/path/to/file.json");
+```
+
+`pantheon_clear_edge_paths` is a platform function that can be implemented in a [Quicksilver hook (like `deploy`)](https://pantheon.io/docs/quicksilver#hooks) to automatically purge a given static asset from cache, for example:
```
Comment (rachelwhitton) [inline source/content/caching-advanced-topics.md:70]: Todo add usage such as 10 max limit, and wildcard paths not supported

**Comment by rachelwhitton** [comment]
Consider including an example that purges a given static asset manually on the fly w/Terminus, for example 

```
terminus drush SITENAME.ENV -- eval "pantheon_clear_edge_paths(['/path/to/file.json'])"
```

File: source/content/caching-advanced-topics.md (hunk 3 above)
Comment (douggreen) [reply source/content/caching-advanced-topics.md:70]: The pantheon_advanced_page_cache module uses this function from a Drupal module, so I don't think this is limited to just Quicksilver hooks.

**Comment by ccharlton** [comment]
The Drush and WP-CLI workarounds are worth publishing in our docs. We get asked this question by developers often enough.

**Comment by EdwardAngert** [comment]
thanks @rachelwhitton @kyletaylored @ccharlton 
I'm moving this to the next step in the docs process so we can get good info into the public docs

**Pull request closed without merging**